xsdvalidate/xsd_validate_activity.go

```diff
@@ -0,0 +1,226 @@
+package xsdvalidate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.artefactual.dev/tools/temporal"
+)
+
+const XSDValidateActivityName = "xsd-validate-activity"
+
+type XSDValidateActivity struct{}
+
+func NewXSDValidateActivity() *XSDValidateActivity {
+	return &XSDValidateActivity{}
+}
+
+type XSDValidateActivityParams struct {
+	// DirectoryPath is the path of the files to be validated.
+	DirectoryPath string
+
+	// Pattern is the regular expression for files to validate.
+	Pattern string
+
+	// XSDFilePathOrURL is the path of the XSD file to use for validation.
+	XSDFilePathOrURL string
```

Comment:
Similar here, if `xmllint` doesn't work with external files, I think the caller should be the one downloading them. XSD files can reference other XSD files, which is not considered in this implementation. So I'd just make a requirement to use local files and only pass the XML and XSD paths to this activity, at least for now.

Reply:
Sounds good!

Reply:
(Fixed.)